### PR TITLE
Adding Thermal to blade, enclosure and rack chassis

### LINKED ISSUE #57: Add Thermal info at BladeChassis
TBD

### LINKED ISSUE #58: Add Thermal info at EnclosureChassis
TBD

### LINKED ISSUE #59: Add Thermal info at RackChassis
TBD

### PR DESCRIPTION
**Pull request by danielreed**
Closes #57, closes #58 and closes #59

---
This change is [Reviewable](https://reviewable.io/reviews/hewlettpackard/oneview-redfish-toolkit/70)

### CODE REVIEW

File: oneview_redfish_toolkit/tests/api/test_blade_thermal.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
```
Comment (llvieira) [inline oneview_redfish_toolkit/tests/api/test_blade_thermal.py]: @danielreed this file would not have the name like **test_thermal.py** and test Thermal class using Enclosure, Blade and Rack? Overall it's a great PR, and I'll accept it as soon as this question to be resolved.

**Comment by llvieira** [comment]
Reviewed 17 of 17 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70)*

**Comment by llvieira** [comment]
Reviewed 17 of 17 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70)*

**Comment by ffdarkpenguin** [comment]
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[oneview_redfish_toolkit/api/blade_chassis.py, line 54 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuSyRzxqyEFjsbVa4qD:-KuSyRzxqyEFjsbVa4qE:b-ukczcf) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/api/blade_chassis.py#L54)):*
> ```Python
>             self.redfish["Links"]["ContainedBy"]["@odata.id"] = "/redfish/v1/Chassis/" \
>                 + server_hardware["locationUri"].split("/")[-1]
>         self.redfish["Thermal"] = collections.OrderedDict()
> ```

This next 3 lines should be in chassis class

---

*[oneview_redfish_toolkit/api/enclosure_chassis.py, line 57 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuSyXzNz8kFMU6T_fx3:-KuSyXzNz8kFMU6T_fx4:bnj7hl0) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/api/enclosure_chassis.py#L57)):*
> ```Python
>         self.redfish["Links"]["ContainedBy"]["@odata.id"] = \
>             "/redfish/v1/Chassis/" + environmental_configuration["rackId"]
>         self.redfish["Thermal"] = collections.OrderedDict()
> ```

same as above

---

*[oneview_redfish_toolkit/api/thermal.py, line 60 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuSzCKazK7Ci94QGMJ7:-KuSzCKazK7Ci94QGMJ8:b-x0ld36) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/api/thermal.py#L60)):*
> ```Python
>             self.redfish["Temperatures"][0]["UpperThresholdCritical"] = \
>                 utilization_dict["metricList"][0]["metricCapacity"]
>             self.redfish["Temperatures"][0]["MinReadingRangeTemp"] = 10
> ```

is this and the below being hard coded ok?

---

*[oneview_redfish_toolkit/mockups/BladeChassis.json, line 25 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuSzp7va9xEoh5e03nE:-KuSzp7va9xEoh5e03nF:b-jzh8r1) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/mockups/BladeChassis.json#L25)):*
> ```JSON
>     "Model": "HPE Synergy 480 Gen9 Compute Module",
>     "IndicatorLED": "Off",
>     "Thermal": {
> ```

Check where its gonna be after moving thermal info to chassis. Also be aware that this was changed on Rack Chassis patch due to missing fields on chassis class

---

*[oneview_redfish_toolkit/mockups/EnclosureChassis.json, line 46 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuTB270UH7mxFKDQDMv:-KuTB271bShp_Tu70kuI:b3nj1hz) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/mockups/EnclosureChassis.json#L46)):*
> ```JSON
>     "Model": "Synergy 12000 Frame",
>     "IndicatorLED": "Lit",
>     "Thermal": {
> ```

Same comment as on BladeChassis

---

*[oneview_redfish_toolkit/tests/blueprints/test_thermal.py, line 143 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuTC2hWUitKoMG2pQAl:-KuTC2hWUitKoMG2pQAm:bu6v4m3) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/tests/blueprints/test_thermal.py#L143)):*
> ```Python
>     #############
>     # Enclosure #
>     #############
> ```

missing tests for enclosure not found and unexect error

---

*[oneview_redfish_toolkit/tests/blueprints/test_thermal.py, line 181 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuTCAm_8Ri0XNfAEmHD:-KuTCAm_8Ri0XNfAEmHE:bnj7hl0) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/tests/blueprints/test_thermal.py#L181)):*
> ```Python
>     ########
>     # Rack #
>     ########
> ```

same as above

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70)*

**Comment by ricardoas** [comment]
I'm not sure of how a RESTful service should be organized and I don't intend this to be a final decision at all... but is there any special reason for this feature not be part of the Chassis blueprint?

---

Reviewed 1 of 17 files at r1.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-:-KuUgQuLKdv4cUeqSI0H:b-w6l0t8)*

**Comment by danielreed** [comment]
I thought I organized it correctly but its a good question.  Its probably one for the team to decide.  There will be other work that will be affected by the decision we make.

---

Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[oneview_redfish_toolkit/api/blade_chassis.py, line 54 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuSyRzxqyEFjsbVa4qD:-KuUi7u-wJZLcyZUQiny:b2kjol0) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/api/blade_chassis.py#L54)):*
<details><summary><i>Previously, ffdarkpenguin wrote…</i></summary><blockquote>

This next 3 lines should be in chassis class
</blockquote></details>

Ok will do.

---

*[oneview_redfish_toolkit/api/enclosure_chassis.py, line 57 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuSyXzNz8kFMU6T_fx3:-KuUiAVgXrMfJAj1NSLP:b2kjol0) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/api/enclosure_chassis.py#L57)):*
<details><summary><i>Previously, ffdarkpenguin wrote…</i></summary><blockquote>

same as above
</blockquote></details>

Ok will do.

---

*[oneview_redfish_toolkit/api/thermal.py, line 60 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuSzCKazK7Ci94QGMJ7:-KuUiDBwVvc9aTAH52ZA:bzf88s) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/api/thermal.py#L60)):*
<details><summary><i>Previously, ffdarkpenguin wrote…</i></summary><blockquote>

is this and the below being hard coded ok?
</blockquote></details>

Yes its hard coded, these values will never change.  They will always be the same for each hardware device in OneView.

---

*[oneview_redfish_toolkit/mockups/BladeChassis.json, line 25 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuSzp7va9xEoh5e03nE:-KuUiNByD9FdVLtoD-02:b32w0x) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/mockups/BladeChassis.json#L25)):*
<details><summary><i>Previously, ffdarkpenguin wrote…</i></summary><blockquote>

Check where its gonna be after moving thermal info to chassis. Also be aware that this was changed on Rack Chassis patch due to missing fields on chassis class
</blockquote></details>

OK

---

*[oneview_redfish_toolkit/mockups/EnclosureChassis.json, line 46 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuTB270UH7mxFKDQDMv:-KuUiOfm5IJEauSJLwVG:b32w0x) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/mockups/EnclosureChassis.json#L46)):*
<details><summary><i>Previously, ffdarkpenguin wrote…</i></summary><blockquote>

Same comment as on BladeChassis
</blockquote></details>

OK

---

*[oneview_redfish_toolkit/tests/api/test_blade_thermal.py, line 1 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuQmk3qlruBIBwyXAFU-r1-1:-KuUiUoE-KMJHfxHDwDe:b5f165e) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/tests/api/test_blade_thermal.py#L1)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

@danielreed this file would not have the name like **test_thermal.py** and test Thermal class using Enclosure, Blade and Rack? Overall it's a great PR, and I'll accept it as soon as this question is resolved.
</blockquote></details>

Are you saying this file should be named test_thermal.py?

---

*[oneview_redfish_toolkit/tests/blueprints/test_thermal.py, line 143 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuTC2hWUitKoMG2pQAl:-KuUi_OZDehjyXNbpH8m:bsldy8c) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/tests/blueprints/test_thermal.py#L143)):*
<details><summary><i>Previously, ffdarkpenguin wrote…</i></summary><blockquote>

missing tests for enclosure not found and unexect error
</blockquote></details>

Do we really need those tests? They all use the same code path that the blade tests covered.

---

*[oneview_redfish_toolkit/tests/blueprints/test_thermal.py, line 181 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuTCAm_8Ri0XNfAEmHD:-KuUirNqSTLTJmuSjH-z:by1ntha) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/tests/blueprints/test_thermal.py#L181)):*
<details><summary><i>Previously, ffdarkpenguin wrote…</i></summary><blockquote>

same as above
</blockquote></details>

Do we really need those tests? They all use the same code path that the blade tests covered.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-:-KuUj7V6vaTAEH2O2FH9:b-76bvbh)*

**Comment by ffdarkpenguin** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[oneview_redfish_toolkit/tests/api/test_blade_thermal.py, line 1 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuQmk3qlruBIBwyXAFU-r1-1:-KuTBgfW7haDejmilnQI:br3v941) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/tests/api/test_blade_thermal.py#L1)):*
<details><summary><i>Previously, danielreed (Daniel Reed) wrote…</i></summary><blockquote>

Are you saying this file should be named test_thermal.py?
</blockquote></details>

Agreed on name and tests

---

*[oneview_redfish_toolkit/tests/api/test_blade_thermal.py, line 2 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuUwZP595RUeVo1nMrj:-KuUwZP595RUeVo1nMrk:b-cjz5g3) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/tests/api/test_blade_thermal.py#L2)):*
> ```Python
> # -*- coding: utf-8 -*-
> ```

yes, the name should be test_thermal.py

---

*[oneview_redfish_toolkit/tests/blueprints/test_thermal.py, line 143 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuTC2hWUitKoMG2pQAl:-KuUx4XyUUwIrED58y2H:bkxp2jj) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/tests/blueprints/test_thermal.py#L143)):*
<details><summary><i>Previously, danielreed (Daniel Reed) wrote…</i></summary><blockquote>

Do we really need those tests? They all use the same code path that the blade tests covered.
</blockquote></details>

I let ricardo decide on that. Closed on my part

---

*[oneview_redfish_toolkit/tests/blueprints/test_thermal.py, line 181 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuTCAm_8Ri0XNfAEmHD:-KuUwvME1AWjO12k60Bd:b2nkwdn) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/tests/blueprints/test_thermal.py#L181)):*
<details><summary><i>Previously, danielreed (Daniel Reed) wrote…</i></summary><blockquote>

Do we really need those tests? They all use the same code path that the blade tests covered.
</blockquote></details>

Ill let ricardo decide that, closing on my part

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70)*

**Comment by ffdarkpenguin** [comment]
I like them apart like Daniel did.

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-:-KuUxLZaleAPn93mcSfa:b-rofh6e)*

**Comment by ffdarkpenguin** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-:-KuUxYKRMjKSq3GA7T1T:bnfp4nl)*

**Comment by llvieira** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[oneview_redfish_toolkit/tests/api/test_blade_thermal.py, line 1 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuQmk3qlruBIBwyXAFU-r1-1:-KuUzAjlL5eTht6Kwk2A:b-9g237d) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/tests/api/test_blade_thermal.py#L1)):*
> Are you saying this file should be named test_thermal.py?
<details><summary><i alt="summary">Quoted 10 lines of code…</i></summary>
>               
>               
>                 
>                   1 hour ago
>                 
>               
>             
>               
>               
>               ffdarkpenguin
</details>
> 
>               
>               
>                 
Yeah, this file would have the name  test_thermal.py, and it will test all chassis resources, not only for blade chassis.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70)*

**Comment by llvieira** [comment]
In my opinion, endpoints that have the same URL prefix ("/redfish/v1/Chassis/") should be the same blueprint. It's really a good question.

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-:-KuV-0wL6izMNDikGrW5:b-6e1e1m)*

**Comment by llvieira** [comment]
:lgtm:

---

Reviewed 5 of 8 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-:-KuVC3-VAI0i9ax7oThe:bnfp4nl)*

**Comment by danielreed** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[oneview_redfish_toolkit/tests/api/test_blade_thermal.py, line 2 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuUwZP595RUeVo1nMrj:-KuVKi01MHrHEhlJ5YIu:b-896fix) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/tests/api/test_blade_thermal.py#L2)):*
<details><summary><i>Previously, ffdarkpenguin wrote…</i></summary><blockquote>

yes, the name should be test_thermal.py
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70)*

**Comment by danielreed** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[oneview_redfish_toolkit/tests/api/test_blade_thermal.py, line 1 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-KuQmk3qlruBIBwyXAFU-r1-1:-KuVKmGPEfIJ-HpNlwXH:b-896fix) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/cc89d3afe36a3f29523fcf29110ccc28931ce9c7/oneview_redfish_toolkit/tests/api/test_blade_thermal.py#L1)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

> Are you saying this file should be named test_thermal.py?
>               
>               
>                 
>                   1 hour ago
>                 
>               
>             
>               
>               
>               ffdarkpenguin
> 
>               
>               
>                 
Yeah, this file would have the name  test_thermal.py, and it will test all chassis resources, not only for blade chassis.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70)*

**Comment by llvieira** [comment]
Reviewed 1 of 8 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70)*

**Comment by ffdarkpenguin** [comment]
Done.

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-:-KuUz3FfvWa3SNRubbuH:b-896fix)*

**Comment by ricardoas** [comment]
:lgtm:

It still seems strange to me making them separate from Chassis... but whatever you guys decided I'm on!!

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/70#-:-KuZybn2Pk68jpf2czeT:b-m7bcwe)*

**Pull request merged**